src/SerilogTracing.Instrumentation.AspNetCore/Instrumentation/AspNetCore/HttpRequestInActivityInstrumentor.cs

```diff
@@ -169,30 +128,71 @@ public void InstrumentActivity(Activity activity, string eventName, object event
         }
     }
 
-    static Activity? RecreateActivity(Activity? activity)
+    internal static Activity? CreateReplacementActivity(Activity? incoming, IncomingTraceParent incomingTraceParent)
+    {
+        return incomingTraceParent switch
+        {
+            // Don't trust the incoming traceparent
+            IncomingTraceParent.Ignore =>
+                // Generate a new root activity, using no information from the traceparent
+                CreateReplacementActivity(incoming, false, false, false, false),
+            
+            // Partially trust the incoming traceparent
+            IncomingTraceParent.Accept =>
+                // Use the propagated trace and parent ids, but ignore any flags or baggage
+                CreateReplacementActivity(incoming, true, true, false, false),
+            
+            // Fully trust the incoming traceparent
+            IncomingTraceParent.Trust =>
+                // The incoming activity is still replaced, so that:
+                // 1. Sampling is properly applied, even if the activity was manually created by ASP.NET Core
+                // 2. Clients that don't send any traceparent header may still produce a recorded activity
+                CreateReplacementActivity(incoming, true, true, true, true),
+            
+            _ => throw new ArgumentOutOfRangeException()
+        };
+    }
+
+    static Activity? CreateReplacementActivity(Activity? incoming, bool inheritTags, bool inheritParent, bool inheritFlags, bool inheritBaggage)
     {
-        Activity.Current = activity?.Parent;
+        var replacement = ReplacementActivitySource.CreateActivity(DefaultActivityName, ActivityKind.Server);
+        if (replacement?.IsAllDataRequested == true)
+        {
+            replacement.ActivityTraceFlags |= ActivityTraceFlags.Recorded;
```

Comment:
We shouldn’t need to do this here; the sampling decision will determine whether the activity should be recorded or not.

Were you noticing a gap in the trace when recorded wasn’t explicitly set?

Reply:
I thought I was - but I'll run through this again and get to the bottom of it - thanks! 👍

Reply:
Sorted out now, it was an `AllData` (instead of `AllDataAndRecorded`) directive being used in the test code 🤦‍♂️